hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,688 @@
+            temp_indices = tuple(temp_indices)
+            out_kwargs[key] = self._cache_dask_chunk_kwargs[key][temp_indices]
+        return out_kwargs
+
+    def __repr__(self):
+        return f"<{self.name}| Iterating == {self._is_iterating}"
+
+    @classmethod
+    def from_signal(
+        cls,
+        signal,
+        key="offsets",
+        signal_axes="metadata",
+        **kwargs,
+    ):
+        """
+        Initialize a marker collection from a hyperspy Signal.
+
+        Parameters
+        ----------
+        signal: BaseSignal
+            A value passed to the Collection as {key:signal.data}
+        key: str
+            The key used to create a key value pair to
+            create the Collection. Passed as {key: signal.data}.
+        collection_class: None or matplotlib.collections
+            The collection which is initialized
+        signal_axes: str, tuple of UniformAxes or None
+            If "metadata" look for signal_axes saved in metadata under .metadata.Peaks.signal_axes
+            and convert from pixel positions to real units before creating the collection. If a tuple
+            of signal axes those Axes will be used otherwise no transformation will
+            happen.
+        """
+        if signal_axes is None or (
+            signal_axes == "metadata"
+            and not signal.metadata.has_item("Peaks.signal_axes")
+        ):
+            new_signal = signal
+        elif signal_axes == "metadata" and signal.metadata.has_item(
+            "Peaks.signal_axes"
+        ):
+            new_signal = signal.map(
+                convert_positions,
+                inplace=False,
+                ragged=True,
+                output_dtype=object,
+                signal_axes=signal.metadata.Peaks.signal_axes,
+            )
+        elif isinstance(signal_axes, (tuple, list)):
+            new_signal = signal.map(
+                convert_positions,
+                inplace=False,
+                ragged=True,
+                output_dtype=object,
+                signal_axes=signal_axes
+            )
+        else:
+            raise ValueError(
+                "The keyword argument `signal_axes` must be one of 'metadata' a"
+                "tuple of `DataAxes` or None"
+            )
+        kwargs[key] = new_signal.data
+        return cls(**kwargs)
+
+    def _get_data_shape(self):
+        for key, item in self.kwargs.items():
+            if is_iterating(item):
+                return item.shape
+        return ()
+
+    def __deepcopy__(self, memo):
+        new_marker = markers2collection(self._to_dictionary())
+        return new_marker
+
+    def _to_dictionary(self):
+        marker_dict = {
+            "marker_type": self.marker_type,
+            "collection_class": self.collection_class.__name__,
+            "plot_on_signal": self._plot_on_signal,
+            "kwargs": self.kwargs,
+        }
+        return marker_dict
+
+    def get_data_position(self, get_static_kwargs=True):
```

Comment:
```suggestion
    def get_current_kwargs(self, get_static_kwargs=True):
```
Is it a more suitable name?

Add docstring for `get_static_kwargs`?

Reply:
What do you think about this?

Reply:
That seems like a a more reasonable name!